src/macros.rs

```diff
@@ -0,0 +1,73 @@
+//! Convenience macros
+
+/// Macro to produce an array of [`VertexAttributeDescriptor`]
+///
+/// Output has type: `[VertexAttributeDescriptor; _]`. Usage is as follows:
+/// ```
+/// let attrs = vertex_attr_array![Float2, Float, Ushort4];
+///
+/// // In case the shader location does not start from 0, it may be specified:
+/// let attrs = vertex_attr_array![2; Float2];
+/// ```
+#[macro_export]
+macro_rules! vertex_attr_array {
+    ($($fmt:ident),*) => {
+        vertex_attr_array!([]; 0; 0; $($fmt,)*)
+    };
+    ($loc:expr; $($fmt:ident),*) => {
+        vertex_attr_array!([]; 0; $loc; $($fmt,)*)
+    };
+    ([$($t:expr,)*] ; $off:expr; $loc:expr;) => { [$($t,)*] };
+    ([$($t:expr,)*] ; $off:expr; $loc:expr; $item:ident, $($more:ident,)*) => {
+        vertex_attr_array!(
+            [$($t,)*
+            $crate::VertexAttributeDescriptor {
+                format: $crate::VertexFormat :: $item,
+                offset: $off,
+                shader_location: $loc,
+            },];
+            $off + vertex_attr_array!(size $item);
+            $loc + 1;
+            $($more,)*
+        )
+    };
+    (size Uchar2) => { 2 };
```

Comment:
should this be a different macro?

Reply:
Possibly yes. Whether it should be visible to users is another matter — perhaps it is useful?

Reply:
let's make it private for now :)